low-level/framework/utils/conf_utils.py

```diff
@@ -167,3 +170,34 @@
 BMC_IP_KEY = "server_node>%s>bmc>ip" % MACHINE_ID
 BMC_USER_KEY = "server_node>%s>bmc>user" % MACHINE_ID
 BMC_SECRET_KEY = "server_node>%s>bmc>secret" % MACHINE_ID
+
+conf_dict = {
+    "site_id": SITE_ID_KEY + "__default:" + DEFAULT_DC,
+    "node_id": NODE_ID_KEY + "__default:" + DEFAULT_SN,
+    "cluster_id": CLUSTER_ID_KEY + "__default:" + DEFAULT_CLUSTER,
+    "rack_id": RACK_ID_KEY + "__default:" + DEFAULT_RACK,
+    "cortx_build": CORTX_BUILD_KEY,
+    "storage_type": STORAGE_TYPE_KEY + "__default:virtual",
+    "server_type": NODE_TYPE_KEY + "__default:virtual"
+}
+
+def fetch_config(conf_list, index='GLOBAL'):
+    try:
+        response = {}
+        default_val = ""
+        for key in conf_list:
+            if key not in conf_dict:
+                raise KeyError(f"key:{key} not found in the config.")
+            output = conf_dict[key]
+            if "__default" in output:
+                req_string, temp_val = output.split("__")
+                default_val = temp_val.split(":")[1]
+            else:
+                req_string = output
```

Comment:
We are supposed to read everytime using confstor api. Why this special case where we have the value directly & using it, for cortx_build key ? I know it wont change, but still we have decided to keep consistent approach to read everything confstor apis everytime, so no component logic would come in between to reflect any config value change, floated by confstor apis in future.

And that way we wont need this '__default' string also, keeping the table simple.

Reply:
@cdeshmukh we are not using value directly for "cortx_build".

> "cortx_build": CORTX_BUILD_KEY
CORTX_BUILD_KEY = "%s>%s" % (RELEASE, TARGET_BUILD) 

this CORTX_BUILD_KEY has the required str to feed in Conf api.
Am I missing something?

Reply:
@cdeshmukh  Removing fetch_config function & directly using conf api as @sandeepsgt suggested.
https://github.com/Seagate/cortx-monitor/pull/381#discussion_r620238312